plugins/woocommerce-admin/client/products/sections/images-section.tsx

```diff
@@ -0,0 +1,191 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	Link,
+	useFormContext,
+	MediaUploader,
+	ImageGallery,
+	ImageGalleryItem,
+} from '@woocommerce/components';
+import { Card, CardBody, DropZone } from '@wordpress/components';
+import { recordEvent } from '@woocommerce/tracks';
+import { useState } from '@wordpress/element';
+import { Product } from '@woocommerce/data';
+import classnames from 'classnames';
+import { Icon, trash } from '@wordpress/icons';
+import { MediaItem } from '@wordpress/media-utils';
+
+/**
+ * Internal dependencies
+ */
+import { ProductSectionLayout } from '../layout/product-section-layout';
+import DragAndDrop from '../images/drag-and-drop.svg';
+import './images-section.scss';
+
+type Image = MediaItem & {
+	src: string;
+};
+
+export const ImagesSection: React.FC = () => {
+	const { getInputProps, setValue } = useFormContext< Product >();
+	const images = ( getInputProps( 'images' ).value as Image[] ) || [];
+	const [ showRemoveZone, setShowRemoveZone ] = useState( false );
+	const [ imageIdToRemove, setImageIdToRemove ] = useState< number >( 0 );
+
+	const getUniqueImages = ( files: Image[] ) => {
+		if ( ! files ) {
+			return [];
+		}
+		files.forEach( ( image: Image ) => {
+			if (
+				image.id &&
+				images.find( ( file ) => file.id === image.id ) === undefined
+			) {
+				images.push( image );
+			}
+		} );
+		setValue( 'images', images );
+		return images;
+	};
+
+	const toggleRemoveZone = () => {
+		setShowRemoveZone( ! showRemoveZone );
+	};
+
+	const setImageToRemove = ( image: string ) => {
+		const tempNode = document.createElement( 'div' );
+		tempNode.innerHTML = image;
+		const imageId = tempNode.getElementsByTagName( 'img' )[ 0 ].id;
+		setImageIdToRemove( parseInt( imageId, 10 ) );
+	};
+
+	const orderImages = ( newOrder: JSX.Element[] ) => {
+		const orderedImages = newOrder.map( ( image ) => {
```

Comment:
Is it possible to just directly call `setValue( 'images', newOrder );`?  Does the data differ pretty significantly?

I'd consider this a shortcoming in the `onOrderChange` callback if so.  Not a blocker for this PR if that's the case, but would love to open an issue to make this easier.

Reply:
That is not possible now. The `onOrderChange` method [returns the items](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/components/src/sortable/sortable.tsx#L80) and we are saving another kind of object under `images`.
I think that it would be a great addition to make the method `onOrderChange` return the same type of object we will use later.